gammapy/modeling/fit.py

```diff
@@ -825,6 +830,35 @@ def write(
             yaml_str = add_checksum(yaml_str)
         path.write_text(yaml_str)
 
+    @classmethod
+    def read(cls, filename, checksum=False):
+        """Read from YAML file.
+
+        Parameters
+        ----------
+        filename : str
+            input filename
+        checksum : bool
+            Whether to perform checksum verification. Default is False.
+        """
+        yaml_str = make_path(filename).read_text()
+        path, filename = split(filename)
+        return cls.from_yaml(yaml_str, path=path, checksum=checksum)
+
+    @classmethod
+    def from_yaml(cls, yaml_str, path="", checksum=False):
+        """Create from YAML string."""
+        data = yaml.safe_load(yaml_str)
+        checksum_str = data.pop("checksum", None)
+        if checksum:
+            yaml_str = yaml.dump(
+                data, sort_keys=False, indent=4, width=80, default_flow_style=False
+            )
+            if not verify_checksum(yaml_str, checksum_str):
+                warnings.warn("Checksum verification failed.", UserWarning)
+
+        return data
```

Comment:
This method is equivalent to `gammapy.utils.scripts.read_yaml`, so if the goal is just to  get a dict we don't need more.

Here .read and .to_yaml  returns a dict not an instance of the class so they are not classmethod.
The write method does not export all the mandatory arguments of OptimizeResult and CovarianceResult __init__ so we cannot really create a read method right now.

Reply:
I would also say that with the covariance bug fixed in  #5317, the need to read back the output of FitResult is less obvious.

Reply:
I agree with not really needing the read of the FitResult, now with all the other updates in the PRs have been made.

I suggest we close?